src/job_description_feature_extraction.py

```diff
@@ -68,3 +70,49 @@ def get_rake_keywords(job_description):
         # save top 3 results, the rest is usually garbage
         keywords.append(rake.run(doc)[:3])
     return keywords
+
+
+def get_top_idf_features(job_description, k):
+    """
+    use TfIdf to extract top k keywords of corpus
+
+    :param job_description: data frame
+    :param k: number of
+    :return: list of k top features
+    """
+    # based on https://stackoverflow.com/questions/25217510/
+    tfidf_vectorizer, term_vector = _vectorize(job_description)
+    # use _build_idf_dict to get a {term: score} dictionary which was the whole point here
+    # but it makes the build fail because of unused variables so yea
+    # idf_dict = _build_idf_dict(tfidf_vectorizer)
+    idf_ranking = np.argsort(tfidf_vectorizer.idf_)[::-1]
+    features = tfidf_vectorizer.get_feature_names()
+    return [features[i] for i in idf_ranking[:k]]
+
+
+def _build_idf_dict(tfidf_vectorizer):
+    """
+    builds a dictionarry of all terms
+
+    :param tfidf_vectorizer
+    :return: dictrionary in form of {term: score}
+    """
+    idf_dict = {}
+    features = tfidf_vectorizer.get_feature_names()
+    # TODO: not sure if that's the most efficient way to do this..
+    for term, score in zip(features, tfidf_vectorizer.idf_):
```

Comment:
I've never been particularly good with `zip`, but you may want to look in to dictionary comprehension - comprehension is usually > for loop - but it depends a lot so `timeit`